src/output.c

```diff
@@ -768,3 +778,55 @@ handle_output_power_manager_set_mode(struct wl_listener *listener, void *data)
 		break;
 	}
 }
+
+void
+virtual_output_add(struct server *server, const char *output_name)
+{
+	if (output_name) {
+		/*
+		 * Prevent creating outputs with the same name
+		 */
+		struct output *output;
+		wl_list_for_each(output, &server->outputs, link) {
+			if (wlr_output_is_headless(output->wlr_output)) {
+				if (!strcmp(output->wlr_output->name, output_name)) {
+					wlr_log(WLR_DEBUG,
+					"refusing to create virtual output with duplicate name");
+					return;
+				}
+			}
+		}
+		strncpy(server->headless.pending_output_name, output_name,
```

Comment:
Can't tell if `strncpy()` goes over the 4096 limit.
Suggest using `snprintf()` for terminating null byte. Could then check return value.

Reply:
It won't go over the 4096 limit because of `sizeof(server->headless.pending_output_name)`. I double checked and `sizeof` returns 4096.

Suggest to keep it like this?

Reply:
Never mind - let's leave it.
You won't know if it's been truncated, but seems pretty unlikely with that size buffer anyway.